### ISSUE TITLE
Syntax error after hlint refactor

### ISSUE THREAD

**Issue by augustss** [haskell/haskell-language-server#1844]
Applying a refactoring from hlint produces a syntactically wrong result.

### Your environment
```sh
myproject$ haskell-language-server --probe-tools
haskell-language-server version: 1.1.0.0 (GHC: 8.10.4) (PATH: C:\Users\lennart.augustsson\AppData\Roaming\cabal\bin\haskell-language-server.exe)
Tool versions found on the $PATH
cabal:          3.4.0.0
stack:          Not found
ghc:            8.10.4
```
Which OS do you use:
Windows

Which lsp-client do you use:
emacs

### Steps to reproduce
Use this file:
```haskell
module M where
import GHC.Stack
foo :: (HasCallStack) => Int
foo = undefined
```
Apply the suggest refactoring to remove redundant parens.

### Expected behaviour
The result should be
```
foo :: HasCallStack => Int
```

### Actual behaviour
The result is
```
foo :: (HasCallStack => Int
```

**Comment by Ailrun** [comment]
Thank you for the great report! I can reproduce the behaviour with your example.
However, I can also reproduce the issue only with `hlint` command line, so it's probably an upstream issue.

**Comment by jneira** [comment]
Upstream issue https://github.com/mpickering/apply-refact/issues/112

**Comment by jneira** [comment]
Hi, `apply-refact` maintainer has fixed the bug and make a release (`0.9.3.0`). As we have `>=0.9` in the .cabal file, updating the index-state in cabal.project and building from master should fix the issue.

**Comment by jneira** [comment]
The issue should be resolved in master, we will have to wait to the next release to close this defintely

**Comment by jneira** [comment]
The fix is included in hls-1.2.0
